main/src/ca/uwaterloo/flix/language/phase/Simplifier.scala

```diff
@@ -93,6 +90,26 @@ object Simplifier {
             val es1 = List(es.head)
             SimplifiedAst.Expression.ApplyAtomic(op, es1, tpe, purity, loc)
 
+          case AtomicOp.Spawn =>
+            // Wrap the expression in a closure: () -> tpe \ eff
+            val List(e1, e2) = es
+            val lambdaTyp = Type.mkArrowWithEffect(Type.Unit, eff, e1.tpe, loc)
+            val fp = SimplifiedAst.FormalParam(Symbol.freshVarSym("_spawn", BoundBy.FormalParam, loc), Ast.Modifiers.Empty, Type.mkUnit(loc), loc)
+            val lambdaExp = SimplifiedAst.Expression.Lambda(List(fp), e1, lambdaTyp, loc)
+            SimplifiedAst.Expression.ApplyAtomic(AtomicOp.Spawn, List(lambdaExp, e2), tpe, Purity.Impure, loc)
+
+          case AtomicOp.Lazy =>
+            // Wrap the expression in a closure: () -> tpe \ Pure
+            val e = es.head
+            val lambdaTyp = Type.mkArrowWithEffect(Type.Unit, Type.Pure, e.tpe, loc)
+            val fp = SimplifiedAst.FormalParam(Symbol.freshVarSym("_lazy", BoundBy.FormalParam, loc), Ast.Modifiers.Empty, Type.mkUnit(loc), loc)
+            val lambdaExp = SimplifiedAst.Expression.Lambda(List(fp), e, lambdaTyp, loc)
+            SimplifiedAst.Expression.ApplyAtomic(AtomicOp.Lazy, List(lambdaExp), tpe, Purity.Pure, loc)
+
+          case AtomicOp.HoleError(_) =>
```

Comment:
What about MatchError then?

Reply:
That's created in Simplifier so it's not a problem :) https://github.com/flix/flix/blob/f63c7e1ed70856545cf42e7bbd2143c792d0ecd0/main/src/ca/uwaterloo/flix/language/phase/Simplifier.scala#L380

Reply:
Again, its the brain that is the problem!